Demos/Cube/Vec3f.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020, Stephan Unverwerth <s.unverwerth@gmx.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+struct Vec3f {
+    Vec3f() = default;
+    Vec3f(float _x, float _y, float _z)
+        : x(_x)
+        , y(_y)
+        , z(_z)
+    {
+    }
+
+    Vec3f operator+(const Vec3f& other) const
+    {
+        return Vec3f(x + other.x, y + other.y, z + other.z);
+    }
+
+    Vec3f operator-(const Vec3f& other) const
+    {
+        return Vec3f(x - other.x, y - other.y, z - other.z);
+    }
+
+    Vec3f operator*(float f) const
+    {
+        return Vec3f(x * f, y * f, z * f);
+    }
+
+    Vec3f operator/(float f) const
+    {
+        return Vec3f(x / f, y / f, z / f);
+    }
+
+    float dot(const Vec3f& other) const
+    {
+        return x * other.x + y * other.y + z * other.z;
+    }
+
+    Vec3f cross(const Vec3f& other) const
+    {
+        return Vec3f(
+            y * other.z - z * other.y,
+            z * other.x - x * other.z,
+            x * other.y - y * other.x);
+    }
+
+    Vec3f normalized() const
+    {
+        float inv_length = 1 / length();
+        return *this * inv_length;
+    }
+
+    void normalize()
+    {
+        float inv_length = 1 / length();
+        x *= inv_length;
+        y *= inv_length;
+        z *= inv_length;
+    }
+
+    float length() const
+    {
+        return sqrt(x * x + y * y + z * z);
+    }
+
+    float x;
+    float y;
+    float z;
```

Comment:
Missing `m_` prefix on members.

Reply:
Can we make an exception for this struct? I would really prefer to write `foo.x` instead of `foo.m_x`. I'll add a comment there concerning the coding style, so nobody gets any silly ideas.

Reply:
The style used throughout the project is something like:

* For types with many methods, prefer class over struct.
* For classes, make public getters and setters, keep members private with `m_` prefix.
* For structs, let everything be public and skip the `m_` prefix.

And we tend to use `foo.x()` rather than `foo.m_x`. :)